flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1270,11 +1266,30 @@ public void stopCheckpointScheduler() {
 		}
 	}
 
+	/**
+	 * Aborts all the pending checkpoints and check whether {@code currentPeriodicTrigger} has been assigned as null.
+	 * If so, we would create a new trigger as {@code currentPeriodicTrigger}.
+	 *
+	 * @param exception The exception.
+	 */
+	public void abortPendingCheckpointsWithTriggerValidation(CheckpointException exception) {
+		synchronized (lock) {
```

Comment:
I could see we're synchronizing on `lock` to prevent starting multiple checkpoint schedulers not only here but also in the `start/stopCheckpointScheduler` methods. IMHO this interferes with the comment on `lock` which indicates it to only safeguard checkpoint updates, or say manipulation of `pendingCheckpoints`. It's kind of dangerous and using a separate lock for scheduler may be more reasonable, but it's not necessary to be improved in this PR.